File: scripts/remove-ganges.ts
import fs from "fs";

const data = JSON.parse(fs.readFileSync("public/data/india-rivers-highres.geojson", "utf-8"));
const initialCount = data.features.length;

const ganges = data.features.filter((f: any) => f.properties.name === "Ganges");
const ganga = data.features.filter((f: any) => f.properties.name === "Ganga");
console.log(`Ganges count: ${ganges.length}`);
console.log(`Ganga count: ${ganga.length}`);

// Remove Ganges
data.features = data.features.filter((f: any) => f.properties.name !== "Ganges");
const finalCount = data.features.length;

console.log(`Removed ${initialCount - finalCount} features named "Ganges"`);

fs.writeFileSync("public/data/india-rivers-highres.geojson", JSON.stringify(data, null, 2));

File: scripts/remove_ganges.py
import json
import sys

try:
    with open('public/data/india-rivers-highres.geojson', 'r') as f:
        data = json.load(f)
        
    initial_count = len(data['features'])
    data['features'] = [f for f in data['features'] if f.get('properties', {}).get('name') != 'Ganges']
    final_count = len(data['features'])
    
    print(f"Removed {initial_count - final_count} features named 'Ganges'")
    
    with open('public/data/india-rivers-highres.geojson', 'w') as f:
        json.dump(data, f)
        
except Exception as e:
    print(e)
